### ISSUE TITLE
TensorFlow2/Segmentation/MaskRCNN Memory issue

### ISSUE THREAD

**Issue by moonh1** [NVIDIA/DeepLearningExamples#753]
Related to **Model/Framework(s)** 
I am clonning / running TensorFlow2/Segmentation/MaskRCNN

**Describe the bug**
I have modified the model bit to implement a method that I am researching, and since I've changed the model structure, I am now trying to train the model from scratch.
I removed PretrainedWeightsLoadingHook from the hook, and ran the script which is 'train_AMP_1GPU.sh'.

I am using V100 gpu which has 32GB mem, but it returns the error below. (although I have enough memory on GPU)

~~~
  (0) Resource exhausted: Out of memory while trying to allocate 3774873600 bytes.
         [[{{node cluster_63_1/xla_run}}]]
Hint: If you want to see a list of allocated tensors when OOM happens, add report_tensor_allocations_upon_oom to RunOptions for current allocation info.

         [[Func/gradients/bn_2/cond_grad/StatelessIf/then/_900/input/_3138/_3431]]
Hint: If you want to see a list of allocated tensors when OOM happens, add report_tensor_allocations_upon_oom to RunOptions for current allocation info.

  (1) Resource exhausted: Out of memory while trying to allocate 3774873600 bytes.
         [[{{node cluster_63_1/xla_run}}]]
Hint: If you want to see a list of allocated tensors when OOM happens, add report_tensor_allocations_upon_oom to RunOptions for current allocation info.
~~~

**To Reproduce**
1. Install / clone repo. and run the training script
2. Setting: I commeted out PretrainedWeightsLoadingHook  part from the function 'get_training_hooks' in distributed_executer.py.

**Environment**
Please provide at least:
* Container version py3.6.9, tensorflow2.2, cuda10.1
* GPUs in the system: V100 32GB
* CUDA driver version (e.g. 418.67): 418.87

Is there any solution to figure out this?
Thank you!

**Comment by jan-golda** [comment]
Hey @moonh1, I've been looking into that but I can't reproduce the problem. Could you please share some more details on how are you running it?

Have you made any changes except commenting out weights loading hook?
Are you using `./scripts/docker/build_tf2.sh` to build the container?

After starting the container it should show you:
```
NVIDIA Release 20.06-tf2 (build 13409669)
TensorFlow Version 2.2.0
```
Is that what you see?

**Comment by moonh1** [comment]
@jan-golda 
Yeah, I used build_tf2.sh for building docker container.
and the lines I removed are,

~~~
    if not MPI_is_distributed() or MPI_rank() == 0:
        training_hooks.append(PretrainedWeightsLoadingHook(
            prefix="resnet50/",
            checkpoint_path=checkpoint_path,
            skip_variables_regex=skip_checkpoint_variables
        ))
~~~
in distributed_executer.py

**Comment by jan-golda** [comment]
@moonh1 thanks!

I'm trying to find the potential cause. Could you please show me the output of running `nvidia-smi` and `env` in the container/machine setup that you are using?

**Comment by moonh1** [comment]
Sure, I have this result with nvidia-smi
![image](https://user-images.githubusercontent.com/32179857/100166392-feb8e380-2eff-11eb-8eff-99234bcf1f38.png)

**Comment by jan-golda** [comment]
Hmm, could you show the rest of the output? On this picture, I see only the header of the table I'm interested in.

**Comment by moonh1** [comment]
Sure, here is whole display I can see with nvidia-smi
![image](https://user-images.githubusercontent.com/32179857/100398533-6e5ed800-3092-11eb-9986-dbd5b0a1568b.png)

thanks!

**Comment by jan-golda** [comment]
So, first of all, I would try updating the drivers, as this might have a big impact.
You currently have `418.87.00` driver while `450.80.02` is available.

https://docs.nvidia.com/datacenter/tesla/index.html

**Comment by jan-golda** [comment]
Hi @moonh1, did you had a chance to try with the newer driver?

**Comment by pribalta** [comment]
Closing this issue due to lack of activity

Please @moonh1 ping us if the problem still persists!
